workers/unity/Assets/Gdk/Core/Error/ConnectionError.cs

```diff
@@ -0,0 +1,20 @@
+namespace Improbable.Gdk.Core
+{
+    public enum ConnectionErrorReason
+    {
+        CannotEstablishConnection,
+        DeploymentNotFound,
+        InvalidConfig
+    }
+
+    public struct ConnectionError : IErrorContext
+    {
+        public string Message;
+        public ConnectionErrorReason Reason;
+
+        public override string ToString()
+        {
+            return Message;
```

Comment:
Do we want to include the reason enum in the string or do we expect the message to encapsulate that information?

Reply:
I'd just have a string field and not bother with the method, but I don't have a strong opinion here. The logger may well add it's own stuff to whatever it outputs.